[OSDOCS-6989] Control plane machine sets for clusters on RHOSP (#66212)
openshift/openshift-docs: modules/cpms-changing-openstack-flavor-type.adoc

// Module included in the following assemblies:
// * machine_management/control_plane_machine_management/cpmso-using.adoc

:_content-type: PROCEDURE
[id="cpms-changing-openstack-flavor-type_{context}"]

> = Changing the {rh-openstack} compute flavor by using a control plane machine set

Reviewer: Are we using the flavor as an example?

Author: Not sure what you mean. Could you rephrase?

Reviewer: I mean we can use other CPMS attributes and the flavor is just an example , right?

Author: @itzikb-redhat It's what the previous writer included in their work--not 100% on it myself. I'm not sure how important the flavor task is specifically, nor if there are limitations on which attributes you can change. @EmilienM ?

Reviewer 2: the flavor example is because CPMS helps with vertical scaling too. You can easily scale your control plane vertically by changing the flavors and obtain better perfs.

Reviewer: Can we state with something like that - The scaling up is possible with CPMS by changing the flavor?

Author: https://github.com/openshift/openshift-docs/pull/66212/commits/67aff27738a7b6429f828d88be92f452c9573d99

You can change the {rh-openstack-first} compute service (Nova) flavor that your control plane machines use by updating the specification in the control plane machine set custom resource.

> In {rh-openstack}, flavors define the compute, memory, and storage capacity of computing instances. 

Author: ```suggestion
In {rh-openstack}, flavors define the compute, memory, and storage capacity of computing instances. By increasing or decreasing the flavor size, you can scale your control plane vertically.
```

.Prerequisites

* Your {rh-openstack} cluster uses a control plane machine set.

.Procedure

. Edit the following line under the `providerSpec` field:
+
[source,yaml]
----
providerSpec:
  value:
    ...

>     flavor: <1>

Author: ```suggestion
    flavor: m1.xlarge <1>
```

----
<1> Specify a larger {rh-openstack}  flavor type with the same base as the previous selection. For example, you can change `m6i.xlarge` to `m6i.2xlarge` or `m6i.4xlarge`.

> . Save your changes.

Reviewer: Is it obvious that after the change the machines will be replaced with ones with a new flavor?

// Module included in the following assemblies:
// * machine_management/control_plane_machine_management/cpmso-using.adoc

:_content-type: PROCEDURE
[id="cpms-changing-openstack-flavor-type_{context}"]
= Changing the {rh-openstack} compute flavor by using a control plane machine set

You can change the {rh-openstack-first} compute service (Nova) flavor that your control plane machines use by updating the specification in the control plane machine set custom resource.

In {rh-openstack}, flavors define the compute, memory, and storage capacity of computing instances. By increasing or decreasing the flavor size, you can scale your control plane vertically.

.Prerequisites

* Your {rh-openstack} cluster uses a control plane machine set.

.Procedure

. Edit the following line under the `providerSpec` field:
+
[source,yaml]
----
providerSpec:
  value:

>     ...

Reviewer 3: For [YAML snips](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#yaml-formatting):
```suggestion
# ...
```

    flavor: m1.xlarge <1>
----

> <1> Specify a {rh-openstack}  flavor type that has the same base as the existing selection. For example, you can change `m6i.xlarge` to `m6i.2xlarge` or `m6i.4xlarge`. You can choose larger or smaller flavors depending on your vertical scaling needs.

Reviewer 3: Extra space
```suggestion
<1> Specify a {rh-openstack} flavor type that has the same base as the existing selection. For example, you can change `m6i.xlarge` to `m6i.2xlarge` or `m6i.4xlarge`. You can choose larger or smaller flavors depending on your vertical scaling needs.
```

providerSpec:
  value:
    ...
    flavor: m1.xlarge <1>
----

> <1> Specify a larger {rh-openstack}  flavor type with the same base as the previous selection. For example, you can change `m6i.xlarge` to `m6i.2xlarge` or `m6i.4xlarge`.

Reviewer: @EmilienM Must it be a larger one ?

Reviewer 2: anything you want. The use case we should highlight is vertical scale. Where you use a bigger flavor or a smaller one is up to your needs.

Reviewer: So maybe it's good to state that because from what I understood from the text it should be larger.

Author: https://github.com/openshift/openshift-docs/pull/66212/commits/67aff27738a7b6429f828d88be92f452c9573d99

Author: ```suggestion
<1> Specify a {rh-openstack}  flavor type that has the same base as the existing selection. For example, you can change `m6i.xlarge` to `m6i.2xlarge` or `m6i.4xlarge`. You can choose larger or smaller flavors depending on your vertical scaling needs.
```

Author: https://github.com/openshift/openshift-docs/pull/66212/commits/5126856bb971640d64e8f432ad85349188fa81f8
